modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/portlet/action/ImportDataDefinitionMVCActionCommand.java

```diff
@@ -117,6 +133,94 @@ private UploadPortletRequest _getUploadPortletRequest(
 				liferayPortletRequest.getPortletName()));
 	}
 
+	private void _uniquifyDataDefinitionFields(DataDefinition dataDefinition) {
+		for (DataDefinitionField dataDefinitionField :
+				dataDefinition.getDataDefinitionFields()) {
+
+			String oldFieldName = dataDefinitionField.getName();
+
+			String newFieldName = _generateFieldName(oldFieldName);
+
+			if (Objects.equals(
+					dataDefinitionField.getFieldType(), "fieldset")) {
+
+				_uniquifyDataDefinitionFieldset(dataDefinitionField);
+			}
+
+			dataDefinitionField.setName((String)newFieldName);
+
+			DataLayout dataLayout = dataDefinition.getDefaultDataLayout();
+
+			for (DataLayoutPage dataLayoutPage :
+					dataLayout.getDataLayoutPages()) {
+
+				for (DataLayoutRow dataLayoutRow :
+						dataLayoutPage.getDataLayoutRows()) {
+
+					for (DataLayoutColumn dataLayoutColumn :
+							dataLayoutRow.getDataLayoutColumns()) {
+
+						String[] dataLayoutColumnFieldNames =
+							dataLayoutColumn.getFieldNames();
+
+						for (int i = 0; i < dataLayoutColumnFieldNames.length;
+							 i++) {
+
+							if (dataLayoutColumnFieldNames[i].equals(
+									oldFieldName)) {
+
+								dataLayoutColumnFieldNames[i] = newFieldName;
+							}
+						}
```

Comment:
I'll look into that, thanks!!

Reply:
@rebsilva, please also take a look on this [article](https://stackabuse.com/search-algorithms-in-java/#binarysearch) before build your search algorithm. :wink: I believe that Stream API in Java is optimized to have a good performance, but the mentioned article is a good reading for you see the importance of a fast search. :slightly_smiling_face:

Reply:
Thank you so much @natocesarrego, leave it to me!

Reply:
Hi @natocesarrego!

About your first question you were right, so I made a separate function to update the `dataLayout` and return as soon the `oldFieldName` is found. Thanks for the heads up!

For the `dataLayoutColumnFieldNames` array question, you are right that the Linear Search is not the best search algorithms in most cases; I looked into the article and documentation that you mentioned and gave it some thought. I have some points that I would like to discuss:

1. This `dataLayoutColumnFieldNames` array seems, for the most cases, to only have **one** value inside it (I couldn't simulate a case where it contains more than one, or think about how this scenario would be); so in this case, the Linear Search is indeed the best. It would be great to know if this `dataLayoutColumnFieldNames` can in reality have a lot of elements for us to decide better which search algorithm we should use.
2. We could use Binary Search, but if the array really don't have much elements, I think it could make things harder than they should. To make the Binary Search we need to sort the array (I assume we would make a copy to sort because since it's in dataLayout I'm not sure if we can change the order of the elements inside the array) and then make the search; if we use the logic from `Arrays.binarySearch` the index returned would be the one from the sorted array and not of the original. I think we can do the Binary search logic somehow on this problem, but do you think that in this case we would still benefit from it?
3. From what I saw about **stream** methods that you mentioned they are amazing to change, filter or map a series of elements (for example, search for all the customers on a list that have an specific nationality and get this list in return); but they can't return the index of a element when they find it. On this case, we really need the index to change the `newFieldName ` exactly where the `oldFieldName` is in the array. We could change the whole array for another one with the new fieldName, but it isn't clear to me yet how to make this with stream.

I'll keep looking for ways to improve this, thanks for your help and support!